api_app/playbooks_manager/controller.py

```diff
@@ -0,0 +1,283 @@
+                        f" Setting to `{DEFAULT_QUEUE}`"
+                    )
+                queue = DEFAULT_QUEUE
+                # get soft_time_limit
+                soft_time_limit = cc.config.soft_time_limit
+                # create task signature and add to list
+                task_signatures.append(
+                    tasks.run_connector.signature(
+                        args,
+                        {},
+                        queue=queue,
+                        soft_time_limit=soft_time_limit,
+                        task_id=task_id,
+                    )
+                )
+
+            except NotRunnableConnector as e:
+                logger.warning(e)
+
+    job.update_analyzers_to_execute(analyzers_used)
+    job.update_connectors_to_execute(connectors_used)
+
+    runner = chord(task_signatures)
+    cb_signature = tasks.post_all_playbooks_finished.signature(
+        [job.pk], immutable=True
+    )
+    runner(cb_signature)
+    return None
+
+
+def job_cleanup(job: Job) -> None:
+    logger.info(f"[STARTING] job_cleanup for <-- {job.__repr__()}.")
+    status_to_set = job.Status.RUNNING
+
+    try:
+        if job.status == job.Status.FAILED:
+            raise AlreadyFailedJobException()
+
+        stats_analyzers = job.get_analyzer_reports_stats()
+        stats_connectors = job.get_connector_reports_stats()
+        stats = {}
+
+        for entry in stats_analyzers:
+            stats[entry] = (
+                stats_analyzers.get(entry, 0) + stats_connectors.get(entry, 0)
+            )
+
+        logger.info(f"[REPORT] {job.__repr__()}, status:{job.status}, reports:{stats}")
+
+        if stats["running"] > 0 or stats["pending"] > 0:
+            status_to_set = job.Status.RUNNING
+        elif stats["success"] == stats["all"]:
+            status_to_set = job.Status.REPORTED_WITHOUT_FAILS
+        elif stats["failed"] == stats["all"]:
+            status_to_set = job.Status.FAILED
+        elif stats["failed"] >= 1 or stats["killed"] >= 1:
+            status_to_set = job.Status.REPORTED_WITH_FAILS
+        elif stats["killed"] == stats["all"]:
+            status_to_set = job.Status.KILLED
+
+    except AlreadyFailedJobException:
+        logger.error(
+            f"[REPORT] {job.__repr__()}, status: failed. Do not process the report"
+        )
+
+    except Exception as e:
+        logger.exception(f"job_id: {job.pk}, Error: {e}")
+        job.append_error(str(e), save=False)
+
+    finally:
+        if not (job.status == job.Status.FAILED and job.finished_analysis_time):
+            job.finished_analysis_time = get_now()
+        job.status = status_to_set
+        job.save(update_fields=["status", "errors", "finished_analysis_time"])
+
+
+def post_all_playbooks_finished(job_id: int) -> None:
+    """
+        Callback fn that is executed after all playbooks have finished.
+    """
+
+    # get job instance
+    job = Job.objects.get(pk=job_id)
+    # execute some callbacks
+    job_cleanup(job)
```

Comment:
Not calling the connectors?

Reply:
Connectors are called automatically by the analyzers at the end of their own execution if I am not wrong?